pkg/types/validators.go

```diff
@@ -25,6 +25,19 @@ func (v Validator) WithRunner(fn ValidateFunc) Validator {
 	return v
 }
 
+// Run applies validation to the given MetaBundle returning a result
+// indicating whether the validation was successful, failed, or encountered
+// an error.
+func (v Validator) Run(mb MetaBundle) ValidatorResult {
+	res := v.Runner(mb)
+
+	res.ValidatorCode = v.Code
```

Comment:
The copying works, but if we ever want to add fields, it's a wide sync accross Validator, Run method and ValidatorResult. I am wondering if this is a temporary solution you implemented? As we had a couple of design discussion on "how to pass validator fields to middlewares"?

Reply:
Yes this is temporary. My ideal state would be to inject Validators with the necessary behavior to populate results. See https://github.com/Ajpantuso/addon-metadata-operator/blob/8bbc42b81b54a2959e8b076c55b191800b5e8c44/pkg/validator/validator.go#L80-L115